thefuck/rules/composer_not_command.py

```diff
@@ -4,13 +4,34 @@
 
 @for_app('composer')
 def match(command):
-    return (('did you mean this?' in command.output.lower()
-             or 'did you mean one of these?' in command.output.lower()))
+    # determine error type
+    # matching "did you mean this" is not enough as composer also gives spelling suggestions for mistakes other than mispelled commands
+    is_undefined_command_error = r"[Symfony\Component\Console\Exception\CommandNotFoundException]" in command.output
+    suggestions_present = (('did you mean this?' in command.output.lower()
+                            or 'did you mean one of these?' in command.output.lower()))
+    return is_undefined_command_error and suggestions_present
 
 
 def get_new_command(command):
-    broken_cmd = re.findall(r"Command \"([^']*)\" is not defined", command.output)[0]
-    new_cmd = re.findall(r'Did you mean this\?[^\n]*\n\s*([^\n]*)', command.output)
-    if not new_cmd:
-        new_cmd = re.findall(r'Did you mean one of these\?[^\n]*\n\s*([^\n]*)', command.output)
-    return replace_argument(command.script, broken_cmd, new_cmd[0].strip())
+    # since the command class already tells us the original argument, we need not resort to regex
+    broken_cmd = command.script_parts[1]
+    one_suggestion_only = 'did you mean this?' in command.output.lower()
+    if one_suggestion_only:
+        new_cmd = re.findall(r'Did you mean this\?[^\n]*\n\s*([^\n]*)', command.output)
+        return replace_argument(command.script, broken_cmd, new_cmd[0].strip())
+    else:
+        # there are multiple suggestions
+        # trim output text to make it more digestable by regex
+        trim_start_index = command.output.find("Did you mean one of these?")
+        short_output = command.output[trim_start_index:]
+        stripped_lines = [line.strip() for line in short_output.split("\n")]
+        # each of the suggested commands can be found from index 1 to the first occurence of blank string
+        try:
+            end_index = stripped_lines.index('')
+        except ValueError:
+            end_index = None
```

Comment:
I'm a little on the fence for this. I wrapped this in a try-except because I know `list.index` raises `ValueError` if the element is not found, however, I also know composer will _always_ pad the output with some blank lines after the list of suggestions. 

I thought about it and decided to _not_ catch the exception – after all, it really is an exceptional case where composer would not output blank lines; it implies that composer has changed the format of its output messages and the exception bubbling to the user can explicitly signal for us to make a new fix for composer. I think it would be preferable over silent failure.